Sources/OpenTelemetryApi/Trace/Status.swift

```diff
@@ -60,3 +60,58 @@ extension Status: CustomStringConvertible {
         }
     }
 }
+
+// this explicit Codable implementation for Status will probably be redundant with Swift 5.5
```

Comment:
I was thinking on keeping the default implementation for swift 5.5 with something like:

```
#if swift(>=5.5)
extension Status: Codable { }
#else 
```
and your implementation in the else, what do you think?

Reply:
I tried this and it works. However, the encoding is different between my explicit encoding and the synthesized encoding. For example, for `AttributeValue`, the synthesized `.string("")` value will be `{"string":""}` with the explicit encoding and `{"string":{"_0": ""}}` with the synthesized encoding. 

This can cause an issue if an app has some signal data persisted pre Swift 5.5, and then it gets updated to a version built post Swift 5.5, and tries to decode the data.

I'm debating what to do:

1. Make the explicit encoding match the synthesized encoding (described [here](https://github.com/apple/swift-evolution/blob/main/proposals/0295-codable-synthesis-for-enums-with-associated-values.md))
2. Keep only the explicit encoding.
3. Make Swift 5.5 the minimal version (currently it's 5.2) and keep only the synthesized encoding.

What would you suggest?

Reply:
Yes, it was my initial concern when I proposed it, but after thinking about it I realized that using Codable does not mean that the output format is directly the codable output, it is just the in-memory representation. The real output format is the one that the user decides with the decoder implemented like JSONEncoder or NSKeyedArchiver

Reply:
My current implementation of the persistence decorator (part of the next submission) already uses JSONEncoder (it's not a user decision). That will result in differences such as the one I gave as an example. 

The persistence decorator will write the encoded data to a file (similar to how the DataDogExporter does it - it is basically copied from it), and it is possible that the data will be read from the file after an app upgrade which may include an update to Swift >= 5.5.

Reply:
Oh, sorry for my confusion. 
Then yes, there will be issues. 
Option 3 is not an option, we should not break existing version just for it.  So the options are 1, 2 or consider the breaking a minimal issue. I would prefer option 1, which means easier maintenance once written for future changes to the classes, but don’t know if the complexity would be too high. What is your preference?

Reply:
Agreed. Option 1 is pretty straight forward to implement. I pushed the changes which include tests for forward compatibility of a pre swift 5.5 encoder with a post swift 5.5 decoder.

Reply:
Thats really great, thanks a lot for the changes.